package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -1017,38 +1070,119 @@ def calc_eucl_distance(a1, a2):
+
+        1. `timeseries[i]`: data for the i-th trajectory frame (at time
+           `timesteps[i]`, see :attr:`timesteps`)
+        2. `timeseries[i][j]`: j-th hydrogen bond that was detected at the i-th
+           frame.
+        3. ``donor_idx, acceptor_idx, donor_index, acceptor_index,
+           donor_name_str, acceptor_name_str, distance, angle =
+           timeseries[i][j]``: structure of one hydrogen bond data item
+
+
+        In the following description, ``#`` indicates comments that are not
+        part of the output::
+
+          results = [
+              [ # frame 1
+                 [ # hbond 1
+                    <donor idx (1-based)>, <acceptor idx (1-based)>, <donor index (0-based)>,
+                    <acceptor index (0-based)>, <donor string>, <acceptor string>,
+                    <distance>, <angle>
+                 ],
+                 [ # hbond 2
+                    <donor idx (1-based)>, <acceptor idx (1-based)>, <donor index (0-based)>,
+                    <acceptor index (0-based)>, <donor string>, <acceptor string>,
+                    <distance>, <angle>
+                 ],
+                 ....
+              ],
+              [ # frame 2
+                [ ... ], [ ... ], ...
+              ],
+              ...
+          ]
+
+        The time of each step is not stored with each hydrogen bond frame but in
+        :attr:`~HydrogenBondAnalysis.timesteps`.
+
+
+        Note
+        ----
+        Each index variable named *idx* is a 1-based index. To get the
+        :attr:`Atom.index` (the 0-based index typically used in MDAnalysis)
+        simply subtract 1, or better, use the 0-based variables named
+        *index*.
+
+        For instance, to find an acceptor atom in :attr:`Universe.atoms` by
+        *index* one would use ``u.atoms[acceptor_index]``.
+
+        The :attr:`timeseries` is a managed attribute and it is generated
+        from the underlying data in :attr:`_timeseries` every time the
+        attribute is accessed. It is therefore costly to call and if
+        :attr:`timeseries` is needed repeatedly it is recommended that you
+        assign to a variable::
+
+           h = HydrogenBondAnalysis(u)
+           h.run()
+           timeseries = h.timeseries
+
+
+        See Also
+        --------
+        :attr:`table` : structured array of the data
+
+
+        .. versionchanged:: 0.16.1
+           :attr:`timeseries` has become a managed attribute and is generated from the stored
+           :attr:`_timeseries` when needed. :attr:`_timeseries` contains the donor atom and
+           acceptor atom specifiers as tuples `(resname, resid, atomid)` instead of strings.
+
+        .. deprecated:: 0.15.0
+           The 1-based indices are being deprecated in favor of zero-based indices
+           and they are targeted for removal in 0.17.0.
+
+        """
+        return [[self._reformat_hb(hb) for hb in hframe] for hframe in self._timeseries]
```

Comment:
The real fix is 
```python
                                frame_results.append(
                                    [h.index + 1, a.index + 1, h.index, a.index,
                                    (h.resname, h.resid, h.name),
                                    (a.resname, a.resid, a.name),
                                    dist, angle])
```
where we store atom information as a tuple `(h.resname, h.resid, h.name)`  and `(a.resname, a.resid, a.name)` instead of a mangled name. These unambiguous data are then stored as `_timeseries`. The `timeseries` is generated to look like the previous output. Importantly, the other methods (`generate_table`, `count_*`, `timeseries_by_type`) now all use the unambiguous data to generate their own data structures. Using unambiguous time series data fixes the issue.

The `_reformat_hb()` is just a way to return the same `timeseries` as in previous versions.

Do you think that this needs to be commented?

(In principle we could just store atom indices